metaflow/plugins/kfp/kfp.py

```diff
@@ -168,123 +193,65 @@ def _get_resource_requirements(node):
                     self.environment,
-                    step_name,
+                    node.name,
                     [step_cli],
-                    task_id,
                 ),
                 total_retries,
+                self._get_resource_requirements(node),
             )
 
-        steps_deque = deque(["start"])  # deque to process the DAG in level order
-        current_task_id = 0
-
-        # set of seen steps, i.e., added to the queue for processing
-        seen_steps = set(["start"])
-        # Mapping of steps to task ids
-        step_to_task_id_map = {}
         # Mapping of steps to their KfpComponent
-        step_to_kfp_component_map = {}
-
-        while len(steps_deque) > 0:
-            current_step = steps_deque.popleft()
-            current_task_id += 1
-            step_to_task_id_map[current_step] = current_task_id
-            current_node = self.graph.nodes[current_step]
-
-            # Generate the correct input_path for each step. Note: input path depends on a step's parents (i.e., in_funcs)
-            # Format of the input-paths for reference:
-            # non-join nodes: "run-id/parent-step/parent-task-id",
-            # branch-join node: "run-id/:p1/p1-task-id,p2/p2-task-id,..."
-            # foreach node: TODO: foreach is not considered here
-            if current_task_id == 1:  # start step
-                # this is the standard input path for the `start` step
-                cur_input_path = "{run_id}/_parameters/0"
-            else:
-                if current_node.type == "join":
-                    cur_input_path = "{run_id}/:"
-                    for parent_step in current_node.in_funcs:
-                        cur_input_path += "{parent}/{parent_task_id},".format(
-                            parent=parent_step,
-                            parent_task_id=str(step_to_task_id_map[parent_step]),
-                        )
-                    cur_input_path = cur_input_path.strip(",")
-                else:
-                    parent_step = current_node.in_funcs[0]
-                    cur_input_path = "{{run_id}}/{parent}/{parent_task_id}".format(
-                        parent=parent_step,
-                        parent_task_id=str(step_to_task_id_map[parent_step]),
-                    )
-
-            step_to_kfp_component_map[current_step] = build_kfp_component(
-                current_node, current_step, current_task_id, cur_input_path
-            )
-
-            for step in current_node.out_funcs:
-                if step not in seen_steps:
-                    steps_deque.append(step)
-                    seen_steps.add(step)
+        task_ids: Dict[str, int] = graph_to_task_ids(self.graph)
+        step_to_kfp_component_map: Dict[str, KfpComponent] = {}
+        for step_name, task_id in task_ids.items():
+            node = self.graph[step_name]
+            step_to_kfp_component_map[node.name] = build_kfp_component(node, task_id)
```

Comment:
What is the difference between the `step_name` and the `node.name` here?

Reply:
none, updating it to `step_name`